lib/flatpak-installation.c

```diff
@@ -1122,6 +1122,16 @@ progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
   guint new_progress = 0;
   gboolean estimating = FALSE;
 
+  /* The heuristic here goes as follows:
+   *  - While we don't know how many files, nor they sizes, grow up to 10%
+   *  - Once we have a minimum number of files from the metadata, we can estimate
+   *    how much we still have to download.
+   *     - If we have extra data to download, this goes up to 45%
+   *     - Else, this goes up to 90%
+   * - Extra data will go from 45% to 90%
+   * - Writing the objects goes from 90% to 100%
```

Comment:
Thats not right. writing the objects happens before the extra data (that is always last as it is not part of the ostree pull operation)